src/frontend/src/handler/create_index.rs

```diff
@@ -159,7 +159,9 @@ pub(crate) fn gen_create_index_plan(
     let (index_database_id, index_schema_id) =
         session.get_database_and_schema_id_for_create(Some(schema_name))?;
 
-    let index_table = materialize.table();
+    let index_table = materialize.table_mut();
+    // Inherit table properties
+    index_table.properties = table.properties.clone();
```

Comment:
Are all table properties applicable to the index table? Will it be better to expose a `set_retention_seconds` method instead of `table_mut` given that we only need to set retention seconds here?